react/components.md

```diff
@@ -0,0 +1,96 @@
+
+### Where do components live
+
+So remember how our component is just hanging out in its own dedicated file? This makes it independent from the rest of the codebase! That said, while independence is great, we do want the component to use functionality created elsewhere, and to share itself with other components. How can we do this? `import`ing and `export`ing! First, compare your `index.js` file to this newly created file, and try to guess what it's missing in order to work. We haven't imported anything! Let's import `React` in order to turn our `Greeting()` function into a functional React component. Up next, we'll want to `export` our newly created component to that parent components can use it as a child throughout your project. 
+
+~~~jsx
+import React from 'react'
+
+function Greeting() {
+  return <h1>"I swear by my pretty floral bonnet, I will end you."</h1>;
+}
+
+export default Greeting;
+~~~
+
+Are we done? Well let's think about this - we're declared our functional component, and exported it, but does `index.js` know about it yet? Nope! Let's fix that. Let's look at `index.js`, we can see that `root.render()` is rendering the `App` component. Let's replace that `App` component with our newly creating greeting, which we'll have to make sure is first imported properly. The end result should look something like this:
+
+~~~jsx
+import React from 'react';
+import ReactDOM from 'react-dom/client';
+import './index.css';
+import Greeting from './greeting.js';
+import reportWebVitals from './reportWebVitals';
+
+const root = ReactDOM.createRoot(document.getElementById('root'));
+root.render(
+  <React.StrictMode>
+    <Greeting />
+  </React.StrictMode>
+);
+
+// If you want to start measuring performance in your app, pass a function
+// to log results (for example: reportWebVitals(console.log))
+// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
+reportWebVitals();
+~~~
+
+And just like that, you've successfully imported and used your first custom-made component, congratulations!
+
+### Assignment
+
+<div class="lesson-content__panel" markdown="1">
+
+1.  Set up another project all on your own, displaying whatever your heart desires
+    *   Use the [Setting Up A React Environment](https://github.com/TheOdinProject/top-meta/issues/221) lesson as a reference
+    *   While components normally get exported as defaults, try using some named exports instead of default exports. If unsure of how to do this, consult your best friend - [MDN its export documentation](https://developer.mozilla.org/en-US/docs/web/javascript/reference/statements/export#description)
+</div>
```

Comment:
Instead of asking to create a new project, we should prompt learners to create new components. For example, create a component to display your favourite food or something.

Reply:
Addressed in [this commit ](https://github.com/TheOdinProject/curriculum/pull/24702/commits/7f49e29bc97e88b7240dd973468d3326a54940ef)